### PR TITLE
Make colors prettier

### PR DESCRIPTION
**Pull request by NoSloppy**

### CODE REVIEW

**Comment by NoSloppy** [comment]
Screenshot 2023-06-04 at 10 11 05 PM
Screenshot 2023-06-04 at 10 11 49 PM
Screenshot 2023-06-04 at 10 12 08 PM

**Comment by profezzorn** [comment]
So if we're making nice looking buttons, why only do it for colors?

**Comment by NoSloppy** [comment]
> So if we're making nice looking buttons, why only do it for colors?

Sure. This is a start?

File: style_editor.html [hunk 1]
```diff
@@ -2636,10 +2651,11 @@
   } else {
     sortString = "A:"+hsl[0];
   }
-  var color = "rgb("+r+","+g+","+b+")";
-  if (r == 0 && g == 0 && b == 0) color = "white";
-  return "<input srt='"+sortString+"' type=button style='border: 1px solid black;padding:8px;color:black;background:"+color+"' class=btn onclick='SetTo(\""+name+"\")' value='"+enc(name)+"'>\n";
+  var text = (name === "Black" || name === "Blue") ? "white" : "black";
```
Comment (profezzorn) [inline style_editor.html:2654]: "text" and "color" needs better name, like maybe textColor and bgColor

File: style_editor.html [hunk 2]
```diff
@@ -2636,10 +2651,11 @@
...
+  var color = (name === "Black") ? "black" : "rgb("+r+","+g+","+b+")";
```
Comment (profezzorn) [inline style_editor.html:2655]: I don't understand why we need a special case for black here.

File: style_editor.html [hunk 3]
```diff
@@ -9629,7 +9645,11 @@
 }
 
 function AddTabContent(tab, data) {
-  FIND(tab + "_tabcontent").innerHTML += data;
+  var tabContentElement = FIND(tab + "_tabcontent");
+  tabContentElement.innerHTML += data;
+  if (tab === 'rgb') {
+    tabContentElement.classList.add('rgb-tabcontent');
```
Comment (profezzorn) [inline style_editor.html:9651]: Why not just add this in the HTML?

File: style_editor.html (hunk 1 above)
Comment (NoSloppy) [reply style_editor.html]: Done.

File: style_editor.html (hunk 2 above)
Comment (NoSloppy) [reply style_editor.html]: changed.

File: style_editor.html (hunk 3 above)
Comment (NoSloppy) [reply style_editor.html:9652]: unsure

File: style_editor.html [hunk 4]
```diff
@@ -1149,8 +1162,10 @@
 function initGL() {
   AddTab("color", "Styles",effect_links.sort().join(""))
   AddTab("rgb", "Colors",
-    "<input type=color id=COLOR value='#ff0000' onclick='ClickColor()' />" +
-    qlinks.sort().join(""));
+    qlinks.sort().join("") +
+    "<div class='custom-color'>" +
+    "<label>Custom color </label>" +
```
Comment (brd55) [inline style_editor.html:1167]: Could you please use the "for" attribute to tie the label to the input?

Comment (NoSloppy) [reply style_editor.html:1167]: good call, thanks.

File: style_editor.html (hunk 3 above)
Comment (profezzorn) [reply style_editor.html:9652]: unsure about what?

Comment (brd55) [reply style_editor.html:9652]: That, or you could target your style to the ID.

Comment (NoSloppy) [reply style_editor.html]: OH I looked right past the fact we're already adding class=tabcontent already right there in AddTab.
But now I'm thinking why target only rgb "Color" tab, when we could just do it for all tabs, leaving the option to style them later if desired. (like the buttons could be styled differently if the tab's buttons have the long strings etc...)
Change applied.

**Comment by profezzorn** [comment]
conflicts

**Comment by NoSloppy** [comment]
> conflicts

resolved

**Pull request merged**